// try-catch
using flx.std.structs;

try {
	print("executed try\n");
	var a = 10/0;
	print("after division");
} catch (var ex: flx::Exception) {
	print("executed catch\n");
	print("generated error: " + ex.error);
}


var i = 0;
try{
  //var i: int = "asda";
  i = 10 / 0;
  print(i);
}catch(var ex: flx::Exception){
  print(ex.error);
}
  print(i);

// var i = 10 / 0;
// print(i);

try {
	print("try\n");
	print(10/0);
} catch (...) {
	print("catch\n");
}

print("end");

